From 0321e7bdb5be3764b0148e617fcb8db296d6877b Mon Sep 17 00:00:00 2001
From: Erik Schmauss <erik.schmauss@intel.com>
Date: Fri, 19 Jan 2018 11:05:58 -0800
Subject: [PATCH] turn on Parse Table as termlist for table loading

---
 include/acpi/acpixf.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/acpi/acpixf.h b/include/acpi/acpixf.h
index c589c3e12d90..0fd609dfb1c4 100644
--- a/include/acpi/acpixf.h
+++ b/include/acpi/acpixf.h
@@ -200,7 +200,7 @@ ACPI_INIT_GLOBAL(u8, acpi_gbl_group_module_level_code, TRUE);
  * a term_list. Default is FALSE, do not execute entire table until some
  * lock order issues are fixed.
  */
-ACPI_INIT_GLOBAL(u8, acpi_gbl_parse_table_as_term_list, FALSE);
+ACPI_INIT_GLOBAL(u8, acpi_gbl_parse_table_as_term_list, TRUE);
 
 /*
  * Optionally use 32-bit FADT addresses if and when there is a conflict
-- 
2.14.3

